pkg/tenantctl/controller.go

```diff
@@ -208,7 +208,7 @@ func (c *Controller) syncHandler(key string) error {
 		if err != nil {
 			return err
 		}
-		tenant, err = c.removeFinalizerAndUpdate(tenant)
+		_, err = c.removeFinalizerAndUpdate(tenant)
```

Comment:
`removeFinalizerAndUpdate` still returns `c.update(tenant)` or `tenant` but you removed it from here. Is this by intention?

Reply:
The err is always returned. The linter also complained about this. Or am I wrong?
https://github.com/SAP/stewardci-core/pull/223/files#diff-1068f0c0ba322e270f1c26f4c86c5cc4b8c7deb0ac3b890ac40af3f5811eb612R215

Reply:
I wrote it due to lack of my go experties. `err` is correct. I meant `tenant` object was used later on after line #211 but it got updated inside `removeFinalizerAndUpdate` function. From my understanding when you pass an object (`tenant` in this case) to a function it does not get passed as value but as reference and when it gets updated inside the function then we don't need to return it from inside the function right? But it got returned from `removeFinalizerAndUpdate`. Now my question is why did we have it like `tenant, err = ...` and not like what you proposed? And why do we need to return tenant from `removeFinalizerAndUpdate` function?

Reply:
1. `tenant` is not used after line 211 since we `return err` in line 215.
2. I think it should be possible to modify the object directly instead of returning a new instance. But since the used client library does it this way we also have to do it this way.

Reply:
Sorry I did not pay attention that we are inside `if` and we return `err`